api/tests-static/test_static_api_with_pointer.cc

```diff
@@ -30,50 +30,77 @@ namespace Awa {
+    EXPECT_EQ(AwaError_DefinitionInvalid,   AwaStaticClient_DefineResource(client_, 300,   1, "TestResource", AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite));
+    EXPECT_EQ(AwaError_DefinitionInvalid,   AwaStaticClient_DefineResource(client_, 7997,  1, "TestResource", AwaResourceType_Opaque, 2, 1, AwaResourceOperations_ReadWrite));
+    EXPECT_EQ(AwaError_DefinitionInvalid,   AwaStaticClient_DefineResource(client_, 7997,  1, NULL,           AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite));
+}
+
+TEST_F(TestStaticClientWithPointerWithServer, AwaStaticClient_SetResourceStorageWithPointer_Invalid)
 {
     AWA_OPAQUE(o, 10);
-    EXPECT_EQ(AwaError_StaticClientInvalid, AwaStaticClient_DefineObject(NULL, "TestObject", 7997, 0, 1));
-    EXPECT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_DefineObject(client_, NULL, 7997, 0, 1));
-    EXPECT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_DefineObject(client_, "TestObject", 7997, 2, 1));
-    EXPECT_EQ(AwaError_Success, AwaStaticClient_DefineObject(client_, "TestObject", 7997, 0, 1)); // valid
-
-    EXPECT_EQ(AwaError_StaticClientInvalid, AwaStaticClient_DefineResourceWithPointer(NULL,  "TestResource", 7997,  1, AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite,  &o, sizeof(o), 0));
-    EXPECT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_DefineResourceWithPointer(client_, "TestResource", 7997,  1, AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite,  &o, 0, 0));
-    EXPECT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_DefineResourceWithPointer(client_, "TestResource", 7997, 1, AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite, NULL, 0, 0));
-    EXPECT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_DefineResourceWithPointer(client_, "TestResource", 300,  1, AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite,  &o, sizeof(o), 0));
-    EXPECT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_DefineResourceWithPointer(client_, "TestResource", 7997,  1, AwaResourceType_Opaque, 2, 1, AwaResourceOperations_ReadWrite, &o, sizeof(o), 0));
-    EXPECT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_DefineResourceWithPointer(client_, NULL, 7997,  1, AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite, &o, sizeof(o), 0));
+    ASSERT_EQ(AwaError_Success,             AwaStaticClient_DefineObject(client_, 7997, "TestObject", 0, 1));
+    ASSERT_EQ(AwaError_Success,             AwaStaticClient_DefineResource(client_, 7997, 1, "TestResource", AwaResourceType_Opaque, 1, 1, AwaResourceOperations_ReadWrite));
+    EXPECT_EQ(AwaError_StaticClientInvalid, AwaStaticClient_SetResourceStorageWithPointer(NULL,  7996,  1, &o, sizeof(o), 0));
+    EXPECT_EQ(AwaError_DefinitionInvalid,   AwaStaticClient_SetResourceStorageWithPointer(client_, 7997,  1, &o, 0, 0));
+    EXPECT_EQ(AwaError_DefinitionInvalid,   AwaStaticClient_SetResourceStorageWithPointer(client_, 7997,  1, NULL, 0, 0));
```

Comment:
Need to test AwaStaticClient_SetResourceStorageWithPointer on a resource that hasn't been defined using AwaStaticClient_DefineResource:

    ASSERT_EQ(AwaError_DefinitionInvalid, AwaStaticClient_SetResourceStorageWithPointer(client_,  7996,  1, &o, sizeof(o), 0));